### PR TITLE
Support PATCH /api/keys on Enabled flag

### PR DESCRIPTION
**Pull request by mkayad**
### Jira link
[FPO-190](https://transformuk.atlassian.net/browse/FPO-190)
### What?

I have added/removed/altered:

- [x] Altered the update endpoint to enable/disable a api key

### Why?

I am doing this because:

- because it does not exist at the moment

### AC

- Tested locally

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,10 +29,29 @@ main.tf to update this to the new table name for all environments (e.g. developm
 
 See [pitr-manual] for how-to instructions on restoring from a Point in Time Recovery snapshot.
 
+## Run locally
+To run the service locally, you will need to install node and yarn. For the first time or anytime, the dependencies in the package.json are updated, we need to install the dependencies as follow:
+```
+yarn install
+
+```
+to start the server, execute the below command
```
Comment (shjohnson) [inline README.md:38]: Do we need to add the `yarn run build` step here?

File: spec/controllers/apiKeysControllerSpec.ts [hunk 2]
```diff
@@ -109,7 +109,7 @@ describe('ApiKeyController', () => {
       }
     } as any
 
-    it('updates the key', async () => {
+    it('disable the key', async () => {
```
Comment (shjohnson) [inline spec/controllers/apiKeysControllerSpec.ts:112]: disables*

File: spec/controllers/apiKeysControllerSpec.ts [hunk 3]
```diff
@@ -136,7 +136,42 @@ describe('ApiKeyController', () => {
...
-        Description: 'description',
+        Description: '',
+        FpoId: '',
+        CreatedAt: apiKey.CreatedAt,
+        UpdatedAt: apiKey.UpdatedAt,
+        UsagePlanId: ''
+      })
+    })
+
+    it('enable the key', async () => {
```
Comment (shjohnson) [inline spec/controllers/apiKeysControllerSpec.ts:147]: enables*

File: spec/controllers/apiKeysControllerSpec.ts [hunk 4]
```diff
@@ -136,7 +136,42 @@ describe('ApiKeyController', () => {
         ApiGatewayId: '',
         Secret: '',
         Enabled: false,
```
Comment (shjohnson) [inline spec/controllers/apiKeysControllerSpec.ts:138]: I wonder if in the test we can ensure the key is enabled to begin with?

File: spec/operations/updateCustomerApiKeySpec.ts [hunk 5]
```diff
@@ -16,17 +16,17 @@ describe('UpdateCustomerApiKey', () => {
     createCustomerApiKey = new UpdateCustomerApiKey(dynamodbClient, apigatewayClient)
   })
 
-  it('updates the api key', async () => {
+  it('disable the api key', async () => {
```
Comment (shjohnson) [inline spec/operations/updateCustomerApiKeySpec.ts:19]: disables*

Comment (shjohnson) [reply spec/operations/updateCustomerApiKeySpec.ts:19]: or maybe its disable's 😂

**Review by shjohnson — COMMENTED** [review]
Couple comments but otherwise looks good! 👍

File: README.md (hunk 1 above)
Comment (willfish) [reply README.md:38]: I think this is replaced now with `make run`

File: README.md [hunk 6]
```diff
@@ -29,10 +29,29 @@ main.tf to update this to the new table name for all environments (e.g. developm
 
 See [pitr-manual] for how-to instructions on restoring from a Point in Time Recovery snapshot.
 
+## Run locally
```
Comment (willfish) [inline README.md:32]: If you're happy I think we can just replace this with what got documented for local development with localstack?

File: README.md [hunk 7]
```diff
@@ -29,10 +29,29 @@ main.tf to update this to the new table name for all environments (e.g. developm
...
+```
+
 ## API Documentation
 
 Once the application is running, open a web browser and navigate to the Swagger UI URL: <http://localhost:5001/api-docs>
 
 This has been disabled to only run in development mode as the endpoints are designed to be be internal.
 
 [pitr-manual]: https://docs.trade-tariff.service.gov.uk/manual/how-to-backup-and-restore-in-aws-dynamodb-pitr.html
+
```
Comment (willfish) [inline README.md:56]: `pre-commit install`

**Comment by willfish** [comment]
We ought to be able to call this PATCH api with curl and have this set the values correctly using our new localstack configuration?

```sh
id=HUBP4NMDNBUKZ168SQTL
curl -X PATCH http://localhost:5001/api/keys/local-development/$id -d '{"enabled":true}' # should now be true in api gateway key and dynamodb table

id=HUBU461PU63PXH7GZFSW
curl -X PATCH http://localhost:5001/api/keys/local-development/$id -d '{"enabled":false}' # should now be false in api gateway key and dynamodb table
```

File: README.md (hunk 1 above)
Comment (mkayad) [reply README.md]: I rolled back my changes on the README file

File: spec/controllers/apiKeysControllerSpec.ts (hunk 2 above)
Comment (mkayad) [reply spec/controllers/apiKeysControllerSpec.ts]: Resolved

File: spec/operations/updateCustomerApiKeySpec.ts (hunk 5 above)
Comment (mkayad) [reply spec/operations/updateCustomerApiKeySpec.ts]: left as disables

File: spec/controllers/apiKeysControllerSpec.ts (hunk 3 above)
Comment (mkayad) [reply spec/controllers/apiKeysControllerSpec.ts]: Fixed the typo

File: README.md (hunk 6 above)
Comment (mkayad) [reply README.md]: yes, I rolledback my changes on the README

**Review by willfish — APPROVED** [review]

**Pull request merged**